WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -492,6 +488,35 @@ private void updateConnectionBar(boolean isConnected) {
         }
     }
 
+    private void handleBlogRemoved() {
+        if (!AccountHelper.isSignedIn()) {
+            ActivityLauncher.showSignInForResult(this);
+        } else {
+            MySiteFragment mySiteFragment = getMySiteFragment();
+            if (mySiteFragment != null) {
+                int allBlogsCount = WordPress.wpDB.getAllBlogsCount();
+                if (allBlogsCount == 0) {
+                    mySiteFragment.setBlog(null);
+                } else {
+                    showSitePickerWithCurrentBlog();
+                }
+            }
+        }
+    }
+
+    private void showSitePickerWithCurrentBlog() {
+        Blog currentBlog = WordPress.getCurrentBlog();
+
+        int currentBlogId;
+        if (currentBlog != null) {
+            currentBlogId = currentBlog.getLocalTableBlogId();
+        } else {
+            currentBlogId = WordPress.wpDB.getAllBlogsIDs().get(0);
```

Comment:
That's totally fair. The reason it doesn't check is because `showSitePickerWithCurrentBlog()` only gets called if `numBlogs() > 0`, but that's not a good reason. I'll change it to a db call like you mentioned.